### PR TITLE
The main script for xcat autotest in jinkens environment

### PR DESCRIPTION
**Pull request by hu-weihua**
This is the main script for xcat autotest in jinkens environment.

This script is triggered by jinkens web page.

The using please refer to ``usage`` function, any comments are welcome

### CODE REVIEW

**Comment by daniceexi** [comment]
@hu-weihua we need a review meeting to discuss code structure, could you schedule one?

**Comment by whowutwut** [comment]
@hu-weihua  Just looking at the file path, i would recommend that we create a directory and put the jenkins related scripts into there so it's self contained. 

==> `xCAT-server/share/xcat/tools/jenkins/``

File: xCAT-server/share/xcat/tools/xcatjktest [hunk 1]
```diff
@@ -0,0 +1,1334 @@
...
+        my $version=$2;
+        if($1 =~ /rhel/i){
+            if($version>6){
+                $netbootv="grub2";
+            }else{
+                $netbootv="yaboot";
+            }
+        }elsif($1 =~ /sles/i){
+            if($version<12){
+                $netbootv="yaboot";
```
Comment (tingtli) [inline xCAT-server/share/xcat/tools/xcatjktest]: It seems for sles11.4, we also use grub2?

File: xCAT-server/share/xcat/tools/xcatjktest [hunk 2]
```diff
@@ -0,0 +1,1334 @@
...
+
+
+#######################################
+# prepare mn
+# copy all necessary file to MN
+######################################
+sub prepare_mn {
+    send_msg(2, "[prepare_mn] start to prepare mn $mn.......");
+
+    if($os =~ /rhels7/i){
```
Comment (tingtli) [inline xCAT-server/share/xcat/tools/xcatjktest]: Since we also support rh6.7, we also need to install these 2 packages on rh6.7

**Comment by hu-weihua** [comment]
@whowutwut @tingtli  @daniceexi  thank very much for your comments.
For below 2 reasons I decide to close this pull request and reopen new one.
1. It's a good option to relocate this script to a new directory just as @whowutwut suggested.
2. After review meeting, we have a lot change.

The new pull request is #900

**Pull request closed without merging**